tests/test_parsing.py

```diff
@@ -1084,3 +1085,60 @@ def test_bind_merge() -> None:
     assert merged_config.flag is True
     assert merged_config.nested1.value == "sun"
     assert merged_config.nested2.value == "cheese"
+
+
+class Color(Enum):
+    RED = "red"
+    BLUE = "blue"
+    GREEN = "green"
+
+
+class Number(Enum):
+    ONE = 1
+    TWO = 2
+    THREE = 3
+
+
+@dataclass
+class EnumEntry:
+    name: str
+    color: Color
+    number: Number
+
+
+def test_enums() -> None:
+    @dataclass
+    class Config:
+        best_colors: list[Color]
+        best_numbers: list[Number]
+        entries: list[EnumEntry]
+
+    with stream_text(
+        """
+        best-colors = ["red", "green", "blue"]
+        best-numbers = [1, 2, 3]
+
+        [[entries]]
+        name = "Entry 1"
+        color = "blue"
+        number = 2
+
+        [[entries]]
+        name = "Entry 2"
+        color = "red"
+        number = 1
+        """
+    ) as stream:
+        config = Binder(Config).parse_toml(stream)
+
+    assert len(config.best_colors) == 3
+    assert len(config.best_numbers) == 3
+    assert config.best_colors.index(Color.RED) == 0
+    assert config.best_colors.index(Color.GREEN) == 1
+    assert config.best_colors.index(Color.BLUE) == 2
+    assert all(num in config.best_numbers for num in Number)
+    assert len(config.entries) == 2
+    assert config.entries[0].color == Color.BLUE
+    assert config.entries[0].number == Number.TWO
+    assert config.entries[1].color == Color.RED
+    assert config.entries[1].number == Number.ONE
```

Comment:
```suggestion
    assert config.entries[0].color is Color.BLUE
    assert config.entries[0].number is Number.TWO
    assert config.entries[1].color is Color.RED
    assert config.entries[1].number is Number.ONE
```
Using `is` for comparing enum members is more idiomatic.

And in the case of an `IntEnum`, it actually makes a difference, because `1 == Number.ONE` evaluates to `True` but `1 is Number.ONE` evaluates to `False`.